docs/development/architecture.org

```diff
@@ -0,0 +1,114 @@
+Nix-Gui may be considered to have two components: the backend and the frontend.
+
+The backend (=nixui/options=) reads information from both NixOS and the configuration path being edited, then converts them to data-structures useful for navigation, search, and editing. The backend maintains in-memory state for all changes and commits those changes to disk by writing to the configuration path.
+
+The frontend (=nixui/graphics=) constructs various widgets comprising Nix-Gui and loads them with data from the backend. The frontend doesn't manage any edit state. All graphical "edits" send an update request to the backend.
+
+* Backend
+The backend handles three major tasks: configuration retrieval to initialize data-structures, in-memory configuration mutation, and configuration commitment. It provides APIs for frontend interaction, but doesn't concern itself with how the frontend renders any of its data.
+
+** Loading the Option Tree Data-Structure
+Nix-Gui queries information from the configuration path and =<nixpkgs>= in order to construct a data-structure (=nixui.options.option_tree.OptionTree=) mapping options and attributes to metadata (=nixui.options.option_tree.OptionData=) including their =NixType=, their =OptionDefinition='s:
```

Comment:
A bit confusing here, what exactly you are saying at the end.

Reply:
Are you asking about `including their =NixType=, their =OptionDefinition='s:`?

This is the python representation of the type and default value of the option.

![image](https://user-images.githubusercontent.com/2560255/144495995-0eb3c221-f0c9-47d8-8a2b-ed1816063a4c.png)

I wonder if including the above screenshot from https://search.nixos.org/options?channel=21.11&show=services.dhcpd&from=0&size=50&sort=relevance&type=packages&query=services.dhcpd would make things more sensible?

Reply:
Please create a separate issue to discuss further.